src/amo/reducers/addonsByAuthors.js

```diff
@@ -1,125 +1,139 @@
+): FetchAddonsByAuthorsAction => {
+  invariant(errorHandlerId, 'An errorHandlerId is required');
+  invariant(addonType, 'An add-on type is required.');
+  invariant(authors, 'Authors are required.');
+  invariant(Array.isArray(authors), 'The authors parameter must be an array.');
 
   return {
-    type: FETCH_OTHER_ADDONS_BY_AUTHORS,
+    type: FETCH_ADDONS_BY_AUTHORS,
     payload: {
       addonType,
       authors,
       errorHandlerId,
-      slug,
+      excludeAddonBySlug,
     },
   };
 };
 
-type LoadOtherAddonsByAuthorsParams = {|
-  slug: string,
+type LoadAddonsByAuthorsParams = {|
   addons: Array<ExternalAddonType>,
+  excludeAddonBySlug?: string,
 |};
 
-type LoadOtherAddonsByAuthorsAction = {|
-  type: typeof LOAD_OTHER_ADDONS_BY_AUTHORS,
-  payload: LoadOtherAddonsByAuthorsParams,
+type LoadAddonsByAuthorsAction = {|
+  type: typeof LOAD_ADDONS_BY_AUTHORS,
+  payload: LoadAddonsByAuthorsParams,
 |};
 
-export const loadOtherAddonsByAuthors = (
-  { addons, slug }: LoadOtherAddonsByAuthorsParams
-): LoadOtherAddonsByAuthorsAction => {
-  if (!slug) {
-    throw new Error('An add-on slug is required.');
-  }
-
-  if (!addons) {
-    throw new Error('A set of add-ons is required.');
-  }
+export const loadAddonsByAuthors = (
+  { addons, excludeAddonBySlug }: LoadAddonsByAuthorsParams
+): LoadAddonsByAuthorsAction => {
+  invariant(addons, 'A set of add-ons is required.');
 
   return {
-    type: LOAD_OTHER_ADDONS_BY_AUTHORS,
-    payload: { slug, addons },
+    type: LOAD_ADDONS_BY_AUTHORS,
+    payload: { addons, excludeAddonBySlug },
   };
 };
 
 type Action =
-  | FetchOtherAddonsByAuthorsAction
-  | LoadOtherAddonsByAuthorsAction;
+  | FetchAddonsByAuthorsAction
+  | LoadAddonsByAuthorsAction;
 
 const reducer = (
   state: State = initialState,
   action: Action
 ): State => {
   switch (action.type) {
-    case FETCH_OTHER_ADDONS_BY_AUTHORS:
+    case FETCH_ADDONS_BY_AUTHORS:
+      if (!action.payload.excludeAddonBySlug) {
+        return state;
+      }
+
       return {
         ...state,
         byAddonSlug: {
           ...state.byAddonSlug,
-          [action.payload.slug]: undefined,
+          [action.payload.excludeAddonBySlug]: undefined,
         },
       };
-    case LOAD_OTHER_ADDONS_BY_AUTHORS:
+    case LOAD_ADDONS_BY_AUTHORS: {
+      const newAddons = { byAddonSlug: {}, byUserId: {} };
+
+      if (action.payload.excludeAddonBySlug) {
+        newAddons.byAddonSlug = {
+          [action.payload.excludeAddonBySlug]: action.payload.addons
+            .slice(0, ADDONS_BY_AUTHORS_PAGE_SIZE)
+            .map((addon) => createInternalAddon(addon)),
+        };
+      }
+
+      const addons = action.payload.addons
+        .map((addon) => createInternalAddon(addon));
+
+      for (const addon of addons) {
+        if (addon.authors) {
+          for (const author of addon.authors) {
+            newAddons.byUserId[author.id] = {
```

Comment:
It looks like you also need a `byUserName` mapping since the input to `fetchAddonsByAuthors()` is a string of usernames (if its param type is correct).

If so, this would be a good opportunity to put the add-ons themselves in a `byId` map and store only add-on IDs in all other mappings. However, fixing the existing `byAddonSlug` mapping would take extra effort. I'd suggest leaving that one as is (with a todo comment) and filing an issue to fix it later. You might as well design the new ones correctly though.

Ultimately, the sagas should all be dispatching an additional [`loadAddonResults()`](https://github.com/mozilla/addons-frontend/blob/5af49b851c3a1808b38fd7cf014a5298480a8d6f/src/core/reducers/addons.js#L84) action so that we can let the add-ons reducer store the actual objects but I think that would require more effort so I wouldn't bother for now.

Reply:
Yeah, it requires more effort and *also* extra HTTP requests. The amount of data we need for where these reducers are used is fine from the search results. It's not hard to add an extra `loadAddonResults` to a saga once we need it, but for now I thought we'd keep the request count down.